common/searchattribute/defs.go

```diff
@@ -81,6 +81,10 @@ const (
 	// define a custom ScheduleId search attribute, in which case the query using the ScheduleId would operate just like
 	// any other custom search attribute.
 	ScheduleID = "ScheduleId"
+
+	// PausedActivityTypes is a search attribute that stores the list of activity types that are paused for a given workflow.
+	// Note: there can be multiple activities with the same type but different activity IDs. In this case there will be only one entry in the list.
```

Comment:
```suggestion
	// PausedActivityTypes stores the list of paused activity types in the workflow.
	// Note: there can be multiple activities with the same type but different activity IDs. In this case there will be only one entry in the list.
```

Suppose there are 2 (or more) activities paused, all of the same type. If the user only resume one of the activities, how are you keeping track if there are still activities paused from the same type?

Reply:
Every time user pause/unpause activity, I will "rebuild" PausedActivityTypes from pending activities. 
In this scenario, if user has 2 paused activities of the same type, and one of them was unpaused:
I will go over pending activities, found those that are paused (there still be one), take its type and then build new PausedActivityTypes.
Since it is the same type, search attribute will not change and we will not update it.

Reply:
Wouldn't it potentially be slow? Imagine a scenario with several workflows with several pending activities.

Reply:
It will not be slow. 
To unpause I need access to mutable state either way. If I have access - it becomes a matter of going through the list of pending activities. Alternative is to parse existing attribute, then remove paused one, then build it back. Alternative is slower.